shared/src/main/scala-2.11-2.12/scala/xml/ScalaVersionSpecific.scala

```diff
@@ -0,0 +1,26 @@
+package scala.xml
+
+import scala.collection.SeqLike
+import scala.collection.generic.CanBuildFrom
+
+private[xml] object ScalaVersionSpecific {
+  import NodeSeq.Coll
+  type CBF[-From, -A, +C] = CanBuildFrom[From, A, C]
+  object NodeSeqCBF extends CanBuildFrom[Coll, Node, NodeSeq] {
+    def apply(from: Coll) = NodeSeq.newBuilder
+    def apply() = NodeSeq.newBuilder
+  }
+}
+
+private[xml] trait ScalaVersionSpecificNodeSeq extends SeqLike[Node, NodeSeq] { self: NodeSeq =>
+  /** Creates a list buffer as builder for this class */
+  override protected[this] def newBuilder = NodeSeq.newBuilder
+}
+
+private[xml] trait ScalaVersionSpecificNodeBuffer { self: NodeBuffer =>
+  override def stringPrefix: String = "NodeBuffer"
+}
+
+private[xml] trait ScalaVersionSpecificIterableSerializable[+A] { // extends Iterable[A] {
+  // protected[this] override def writeReplace(): AnyRef = this
+}
```

Comment:
Adding this null trait for 2.11 and 2.12 causes a strange error:

```
> xml/testOnly scala.xml.PatternMatching
[error] Test scala.xml.PatternMatching.nodeSeq failed: expected:<<a/><b/>> but was:<List(<a/>, <b/>)>, took 0.047 sec
[error] Failed: Total 6, Failed 1, Errors 0, Passed 5
```

Reply:
old `SafeNodeSeq.unapplySeq` returns `List`, but new `SafeNodeSeq.unapplySeq` returns `NodeSeq`.


`NodeSeq == List` always return `false`.

```
$ scala -language:_
Welcome to Scala 2.12.7 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> import scala.xml._
import scala.xml._

scala> :paste
// Entering paste mode (ctrl-D to finish)

  object SafeNodeSeq {
    def unapplySeq(any: Any): Option[collection.immutable.Seq[Node]] = any match {
      case s: Seq[_] => Some((s flatMap (_ match {
        case n: Node => n case _ => NodeSeq.Empty
      })).toSeq) case _ => None
    }
  }

// Exiting paste mode, now interpreting.

defined object SafeNodeSeq

scala> SafeNodeSeq.unapplySeq(List(<a/>)).get.getClass.getSuperclass
res0: Class[?0] forSome { type ?0 >: T; type T <: scala.collection.immutable.Seq[scala.xml.Node] } = class scala.xml.NodeSeq

scala> :paste
// Entering paste mode (ctrl-D to finish)

  object SafeNodeSeq {
    def unapplySeq(any: Any): Option[Seq[Node]] = any match {
      case s: Seq[_] => Some(s flatMap (_ match {
        case n: Node => n case _ => NodeSeq.Empty
      })) case _ => None
    }
  }

// Exiting paste mode, now interpreting.

defined object SafeNodeSeq

scala> SafeNodeSeq.unapplySeq(List(<a/>)).get.getClass
res1: Class[_ <: Seq[scala.xml.Node]] = class scala.collection.immutable.$colon$colon
```

Reply:
Actually, it turns out that the trait has nothing to do with the issue.  Merging the 2.13.0-M4 fixes and leaving the 2.13.0-M5 changes helped to isolate the issue.